### PR TITLE
User list

### PR DESCRIPTION
**Pull request by jersonlopez**
## Summary

list all user and verify that user is loged

## Trello Card

[https://trello.com/c/oHAjebh7/4-lista-de-usuarios](https://trello.com/c/oHAjebh7/4-lista-de-usuarios)

### CODE REVIEW

File: app/controllers/user.js [hunk 1]
```diff
@@ -71,3 +68,25 @@ exports.singIn = async (req, res, next) => {
     next(err);
   }
 };
+
+exports.userList = async (req, res, next) => {
+  const query = req.query
+    ? {
+        page: req.query.page,
+        count: req.query.count
```
Comment (picollomartin) [inline app/controllers/user.js]: Add some default values, like page `0` and count `10`

File: app/middlewares/auth.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const auth = req.headers[AUTHORIZATION];
+
+  if (auth) {
+    const user = decoder(auth);
+    const result = await User.getUserBy(user.email);
+
+    if (result != null) {
+      next();
+    } else {
+      res.status(401);
```
Comment (picollomartin) [inline app/middlewares/auth.js]: Add an error for this, use `next(errors.myError)`

File: app/middlewares/auth.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      res.end();
+    }
+  } else {
+    res.status(401);
```
Comment (picollomartin) [inline app/middlewares/auth.js]: Same here

File: test/user.js [hunk 4]
```diff
@@ -123,95 +132,149 @@ describe('users', () => {
...
       chai
         .request(server)
-        .post('/users/sessions/')
-        .send({
+        .get('/users?count=1&page=1')
+        .catch(err => err.should.have.status(401))
+        .then(() => done());
+    });
+
+    it('should fail list all users by pagination because page number is no sent sent', done => {
```
Comment (picollomartin) [inline test/user.js]: Check typo, this error is not mandatory

File: test/user.js [hunk 5]
```diff
@@ -123,95 +132,149 @@ describe('users', () => {
...
+              err.response.body.should.have.property('message');
+              err.response.body.should.have.property('internal_code');
+              expect(err.response.body.internal_code).to.equal('query_error');
+              done();
+            });
         });
+      });
+    });
+
+    it('should fail list all users by pagination because count is no sent sent', done => {
```
Comment (picollomartin) [inline test/user.js]: Same here

File: test/user.js [hunk 6]
```diff
@@ -123,95 +132,149 @@ describe('users', () => {
...
-          expect(err.response.body.internal_code).to.equal('signin_error');
-          done();
+        }).then(res => {
+          chai
+            .request(server)
+            .get('/users?count=1&page=1')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .then(result => {
+              result.should.have.status(200);
+              expect(result).to.be.a('object');
```
Comment (picollomartin) [inline test/user.js:220]: Check if pagination is working, for example count of results

**Review by picollomartin — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by picollomartin — CHANGES_REQUESTED** [review]
My bad, picollomartin is my personal account :disappointed:

**Review by picollomartin — APPROVED** [review]

**Pull request merged**